api/app/signals/apps/signals/managers.py

```diff
@@ -552,17 +552,20 @@ def _update_user_signal_no_transaction(self, data, signal):
         from signals.apps.signals.models.signal_user import SignalUser
         from signals.apps.users.models import User
 
+        created_by = data['user_assignment']['created_by'] if 'created_by' in data['user_assignment'] else None
+        assigned_user_email = data['user_assignment']['user']['email'] if data['user_assignment'] else None
         try:
-            user_email = data['user_assignment']['user']['email']
-            signal.user_assignment, _ = SignalUser.objects.get_or_create(
-                _signal=signal,
-                user=None if not user_email else User.objects.get(email=user_email),
-                created_by=data['created_by'] if 'created_by' in data else None
-            )
-            signal.save()
-        except Exception:
+            assigned_user = User.objects.get(email=assigned_user_email) if assigned_user_email else None
+        except User.DoesNotExist:
```

Comment:
Raise with a different error message that is more meaningfull. Something like: `raise ValidationError('Could not set user assignment, user does not exists')`